### PR TITLE
SSH: Add local TCP forwarding option

### PR DESCRIPTION
**Pull request by adamgreg**
Add allow_tcp_forwarding option, to allow local port forwarding by the SSH add-on. Code lifted from the more advanced [SSH & Web Terminal](https://github.com/hassio-addons/addon-ssh) add-on.

Motivating use case: I wanted to use the ptvsd integration to help debug an issue on a remote Home Assistant instance I'm supporting. For this I would like to use port forwarding via the SSH add-on. It would also be useful for accessing the host SSH server without having to modify the router configuration to forward port 22222.

### CODE REVIEW

File: ssh/config.json [hunk 1]
```diff
@@ -22,11 +22,13 @@
   "map": ["config:rw", "ssl:rw", "addons:rw", "share:rw", "backup:rw"],
   "options": {
     "authorized_keys": [],
-    "password": ""
+    "password": "",
+    "allow_tcp_forwarding": false
```
Comment (pvizeli) [inline ssh/config.json:26]: I think somethings like this would be better:
```yaml
server:
  tcp_forwarding: false
```

Comment (adamgreg) [reply ssh/config.json:26]: Sure, sounds good to me. I just copied the option as-is from the SSH & Web terminal add-on.

Comment (adamgreg) [reply ssh/config.json:26]: I just noticed the `server:` key in your comment. Is the extra level of hierarchy necessary? All of the options are for the SSH server.

Comment (pvizeli) [reply ssh/config.json:26]: yes

**Pull request merged**